src/main/java/homer/controller/impl/electricalmeter/ElectricalMeterImpl.java

```diff
@@ -63,13 +65,20 @@ public void removeOutlet(final Outlet outlet) {
 
     @Override
     public void computeConsumption() {
-        this.globalConsumption = outlets.stream()
-                .mapToDouble(Outlet::getState)
-                .sum();
+        double cons = 0.0;
+        for (Outlet outlet : this.getOutlets()) {
+            cons += ((OutletState) outlet.getState()).getPower().get();
+        }
+        this.globalConsumption = cons;
```

Comment:
How can I implement this using Streams?

Reply:
Probly something like
this.getOutlets().stream().map(e -> e.getState()).map(e -> e.getPower())

then use filter and map again to get the values, then turn it into a DoubleStream and you can use .sum()